[5.1] doc-serialization: exclude non-public decls when serializing module group info.

nkcsgexi opened the pull request:
This reduces the size of x86_64.swiftdoc about 10%.

rdar://48689311

nkcsgexi committed:
doc-serialization: exclude non-public decls when serializing module group info.

This reduces the size of x86_64.swiftdoc about 10%.

rdar://48689311

nkcsgexi commented:
@swift-ci please test

nkcsgexi commented:
@swift-ci Please clean test Linux platform

akyrtzi commented on test/Serialization/group_info_diags.swift:
```diff
@@ -4,7 +4,16 @@
 
-public protocol P {}
+// RUN: %target-swift-frontend -emit-module %s -module-name HasArray -o %t -module-cache-path %t/mcp -group-info-path %S/Inputs/group_info.json -emit-module-doc &> %t/result.txt
+// RUN: strings %t/HasArray.swiftdoc > %t/doc_strings.txt
```
@shahmishal @compnerd should use of `strings` binary in the test be guarded with `// REQUIRES: shell` ?

compnerd replied:
No, you should use `llvm-strings` instead.

akyrtzi replied:
Ah nice, I wasn't aware of this, @nkcsgexi could you make such a change ?

nkcsgexi replied:
sure, I have cherry-picked @compnerd 's https://github.com/apple/swift/pull/24092 here

compnerd committed:
test: repair the Windows test suite

`strings` is not available, use `%llvm-strings` which uses the LLVM
`strings` implementation instead.

nkcsgexi commented:
@swift-ci please test
